### PR TITLE
Build: Fix minor build warning related to link tag.

### PR DESCRIPTION
**Pull request by nk1506**
Before:
Screenshot 2024-02-28 at 5 06 37 PM

After 
Screenshot 2024-02-28 at 5 00 27 PM

### CODE REVIEW

**Comment by nk1506** [comment]
CC: @nastra , @ajantha-bhat

**Comment by nastra** [comment]
@nk1506 do you maybe have the warning that indicated an issue with the link?

File: spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java [hunk 1]
```diff
@@ -201,7 +201,7 @@ TableCatalog tableCatalog() {
 
   /**
    * Closes this procedure's executor service if a new one was created with {@link
-   * #executorService(int, String)}. Does not block for any remaining tasks.
+   * BaseProcedure#executorService(int, String)}. Does not block for any remaining tasks.
```
Comment (ajantha-bhat) [inline spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java:204]: And can you also fix for other spark versions aswell?

**Review by nastra — APPROVED** [review]
LGTM, thanks for fixing

File: spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java (hunk 1 above)
Comment (ajantha-bhat) [reply spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java:204]: I haven't compiled myself. But why it is not giving the same warnings for other javadocs in the same class, like `#withIcebergTable`?

**Pull request merged**